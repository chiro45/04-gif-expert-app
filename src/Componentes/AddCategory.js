import React, {useState} from 'react'
import PropTypes from 'prop-types';
export default function AddCategory({setCategories}) {
    
    const [inputValue, setinputValue] = useState("");
    const handleInputChange = (e)=>{
        setinputValue(e.target.value)
    }
    const handleSubmit = (e)=>{
        e.preventDefault(e)
        console.log('handle submit', inputValue)

        if(inputValue.trim().length > 2){
            
         setCategories( categories => [inputValue, ...categories,])
         setinputValue('')

        }
        //console.log('submit hecho')
    }
    return (      
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
        <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        />
        </form>   
    )
}
AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}