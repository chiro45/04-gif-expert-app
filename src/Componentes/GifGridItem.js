import React from 'react'
import PropTypes from 'prop-types';
export const GifGridItem = ({ title, img}) => {
    
    return (
        <div className='card animate__animated animate__backInRight'>
            <img src={img} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,

}