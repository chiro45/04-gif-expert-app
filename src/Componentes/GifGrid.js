import React from 'react'
import { useFethGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {
    const {data:images, loading} = useFethGifs(category);
    return (
        <>
        <h3 animate__animated animate__backInRight>{category}</h3>
        {loading && <p className='loading animate__animated animate__backInLeft '>Loading...</p>}
        <div className='card-grid animate__animated animate__backInRight'>   
                {
                    images.map( img =>(
                       <GifGridItem
                       key={img.id}
                       {...img}
                       />
                    ))
                }
        </div> 
        </>
    )
}
GifGrid.propTypes ={
    category: PropTypes.string.isRequired
}