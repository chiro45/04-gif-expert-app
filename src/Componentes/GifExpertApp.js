import React, {useState} from 'react';
import AddCategory from './AddCategory';
import PropTypes from 'prop-types';
import { GifGrid } from './GifGrid';


export const GifExpertApp = () =>{
    
    const [categories, setCategories] = useState( ['Avengers']);
    // const handleAdd = ()=>{
    //    setcategories([...categories, "juan CArlo"])
    // }
    
    // const categories = ['one punch', 'samurai X', 'dragon Ball'];
    return ( 
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories ={setCategories}/>
        <hr/>
        <ol>

        {

        categories.map(category => (
                       < GifGrid 
                       key={category}
                       category={category}/>
                    )
        
        )}

        </ol>
        
        </>
     );

    
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


