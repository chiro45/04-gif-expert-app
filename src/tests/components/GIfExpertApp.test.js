
import React from 'react';
const { shallow } = require("enzyme")
import {GifExpertApp} from '../../Componentes/GifExpertApp'



describe('pruebas sobre el componente <GifExpertApp/>', () => {
    test('mostrar el componente correctamente', () => {
        const wrapper = shallow(<GifExpertApp/>)   
        expect(wrapper).toMatchSnapshot(); 
    
    })
    test('debe de mostrar una lista de categorias', () => {
        const categories =['one punch', ' dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
    
})
