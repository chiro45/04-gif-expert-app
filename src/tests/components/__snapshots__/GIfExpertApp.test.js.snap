// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el componente <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category=" dragon Ball"
      key=" dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`pruebas sobre el componente <GifExpertApp/> mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
